pandas/core/reshape/melt.py

```diff
@@ -32,7 +32,11 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
             raise ValueError('id_vars must be a list of tuples when columns'
                              ' are a MultiIndex')
         else:
+            # Check that `id_vars` are in frame
             id_vars = list(id_vars)
+            missing = [v for v in id_vars if v not in frame.columns]
```

Comment:
can be `frame.columns.difference(id_vars)` I think.

Reply:
Referencing the example in #23570:
```python
id_vars = ['Name', 'day']
df.columns.difference(id_vars) # --> ['burgers', 'fries']
```
This is not the intended output though. Here I'm collecting passed column names that aren't in the frame's columns (is each value of `id_vars` in `frame.columns`, **NOT** are there other columns in the frame besides `id_vars`).

Reply:
Got it backwards. `Index(id_vars).difference(frame.columns)`?